Projects/Application/Sources/Common/Network/Http/Request.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Created by Maxim Berezhnoy on 2019-02-12.
+// Licensed under the MIT license
+//
+//
+
+import Foundation
+
+extension Http {
+    final class Request {
+        let urlRequest: URLRequest
+
+        init(from urlRequest: URLRequest) {
+            self.urlRequest = urlRequest
+        }
+
+        func then(completionHandler: @escaping (Response) -> Void) {
```

Comment:
I think that `then` and `cancel` are not parts of the request, but parts of a job over this request. So maybe it should be a part of `Http` class or part of a new type which represents request job like `URLSessionTask`

Reply:
The idea is that `Request` itself encapsulated the job. It is not responsible for anything else except for managing the job lifetime and delivering response.

Reply:
As per other comment, `Request` was renamed to `Task` to clarify responsibility.